core/Plugin/Dimension/ActionDimension.php

```diff
@@ -38,111 +35,8 @@
-    );
-    ```
-     * @api
-     */
-    public function install()
-    {
-        if (empty($this->columnName) || empty($this->columnType)) {
-            return array();
-        }
-
-        return array(
-            $this->tableName => array("ADD COLUMN `$this->columnName` $this->columnType")
-        );
-    }
-
-    /**
-     * Updates the action dimension in case the {@link $columnType} has changed. The update is already implemented based
-     * on the {@link $columnName} and {@link $columnType}. This method is intended not to overwritten by plugin
-     * developers as it is only supposed to make sure the column has the correct type. Adding additional custom "alter
-     * table" actions would not really work since they would be executed with every {@link $columnType} change. So
-     * adding an index here would be executed whenever the columnType changes resulting in an error if the index already
-     * exists. If an index needs to be added after the first version is released a plugin update class should be
-     * created since this makes sure it is only executed once.
-     *
-     * @return array An array containing the table name as key and an array of MySQL alter table statements that should
-     *               be executed on the given table. Example:
-     * ```
-    array(
-    'log_link_visit_action' => array("MODIFY COLUMN `$this->columnName` $this->columnType", "DROP COLUMN ...")
-    );
-    ```
-     * @ignore
-     */
-    public function update()
-    {
-        if (empty($this->columnName) || empty($this->columnType)) {
-            return array();
-        }
-
-        return array(
-            $this->tableName => array("MODIFY COLUMN `$this->columnName` $this->columnType")
-        );
-    }
-
-    /**
-     * Uninstalls the dimension if a {@link $columnName} and {@link columnType} is set. In case you perform any custom
-     * actions during {@link install()} - for instance adding an index - you should make sure to undo those actions by
-     * overwriting this method. Make sure to call this parent method to make sure the uninstallation of the column
-     * will be done.
-     * @throws Exception
-     * @api
-     */
-    public function uninstall()
-    {
-        if (empty($this->columnName) || empty($this->columnType)) {
-            return;
-        }
-
-        try {
-            $sql = "ALTER TABLE `" . Common::prefixTable($this->tableName) . "` DROP COLUMN `$this->columnName`";
-            Db::exec($sql);
-        } catch (Exception $e) {
-            if (!Db::get()->isErrNo($e, '1091')) {
-                throw $e;
-            }
-        }
-    }
-
-    /**
-     * Get the version of the dimension which is used for update checks.
-     * @return string
-     * @ignore
-     */
-    public function getVersion()
-    {
-        return $this->columnType;
-    }
+    protected $dbTableName = 'log_link_visit_action';
+    protected $category = 'General_Actions';
```

Comment:
Simplified code for `ActionDimension`, `VisitDimension`, `ConversionDimension` by having the same logic for all dimensions in `Dimension` class